crates/spirv-std/src/ray_tracing.rs

```diff
@@ -159,3 +159,493 @@ bitflags::bitflags! {
         const SKIP_AABBS = 512;
     }
 }
+
+/// A ray query type which is an opaque object representing a ray traversal.
+#[spirv(ray_query)]
+pub struct RayQuery {
+    _private: u32,
+}
+
+/// Constructs an uninitialized ray query variable. Using the syntax
+/// `let (mut)? <name>`. Where `name` is the name of the ray query variable.
+#[macro_export]
+macro_rules! ray_query {
+    (let $name:ident) => {
+        $crate::ray_query!(@inner $name)
+    };
+    (let mut $name:ident) => {
+        $crate::ray_query!(@inner $name, mut)
+    };
+    (@inner $name:ident $(, $mut:tt)?) => {
+        #[allow(deprecated)]
```

Comment:
You shouldn't need `#[allow(deprecated)]` if you're not calling `mem::uninitialized`. Also, you can move this into the `unsafe` block, and even change its type to `&$($mut)? RayQuery`.

Or even just initialize the user variable with `asm!`, no need for 2 variables.